.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
+- repo: local
+  hooks:
+  - id: must-match-hash
+    name: bison-parser-hashcheck
+    language: python
+    entry: python hooks/must-match-hash.py
+    files: 'src/idl/src/parser\.y$'
+  - id: checked-in-together
+    name: bison-parser-all-files-checked
+    language: python
+    entry: python hooks/checked-in-together.py
+    pass_filenames: false
+    args: ['src/idl/src/parser.c', 'src/idl/src/parser.h', 'src/idl/src/parser.y', 'src/idl/src/parser.y.hash']
```

Comment:
Nitpicking a bit here, but what if `parser.y` changes in a way that doesn't affect the output (e.g., fixing a comment)? I reckon that'll be a rare event, so as long as there is a good way around it, that'd be fine.

Reply:
Solved by appending the parser.y hash to parser.{c,h} in a comment at cmake configure time.